### PR TITLE
Brings Cryopods up to /tg/ codebase standards, makes it drop all items on player deletion & can only be self-used unless your target is catatonic.

### LINKED ISSUE #58402: The global list of theft items is only populated when a steal objective is initialized, which may be problematic for cryopods.
## Round ID:

Found on local

## Reproduction:

https://github.com/tgstation/tgstation/blob/507a82660466fb329ad07a63f6a77fd11abc22d8/code/game/gamemodes/objective.dm#L558-L572

The global list of steal objective items, `GLOB.possible_items`, is only populated on `/datum/objective/steal/New()`. This means the list is empty if no one has a steal objective. 

https://github.com/tgstation/tgstation/blob/507a82660466fb329ad07a63f6a77fd11abc22d8/code/game/machinery/cryopod.dm#L263-L267

Cryopods use this list to check which items to preserve, meaning if there are no antagonists with steal objectives, no items are preserved and high-risk items are deleted.

### PR DESCRIPTION
**Pull request by Timberpoes**
## About The Pull Request

The code has overall had a minor touch-up with a few renamed vars here and there for consistency's sake. The message template for informing the admins before you log out has been hoisted up to a #define.

You can no longer put other players into cryopods unless they're catatonic. This is because cryopods actually ROUND REMOVE people and going SSD is not permanent. An SSD player is free to return to the game at any time and has full metaprotections against being killed while SSD as if they were still living.

image
image

With this removed, the only way to enter pods when a body has a ckey associated with it is of your own volition. As a result, the 3 minute timer is no longer necessary and the only timer is 30 seconds. After this point, you will be round removed if you go SSD or ghost yourself. This timer now uses our COOLDOWN_ macros.

The alerts now use tgalert.
vJxCMESklA

These alerts for vital station crew and antags now always show if you're going to try and cryopod yourself, there was no point giving you a 5 minute cooldown where the alerts don't appear.

If you have a mind.assigned_role that doesn't have a job datum assigned to it, the pod no longer runtimes and you can successfully cryo yourself. Useful for getting rid of your Centcom intern hordes after you've successfully besieged the station.

All items you had equipped (and maybe even some you didn't have equipped) are dumped to the floor when the machine vores you. Items are no longer destroyed, including any special objectives. The cryo console no longer has the functionality to store items inside it, as this functionality has been replaced by the classic floordump. This fixes #58402

EnkGInYnCs

Because the item functionality is no longer needed/supported, the tgui interface has has this section removed. It now just shows a list of all stored crew.

image
image

## Why It's Good For The Game

Aligns cryopods more with the standards and expectations of the tg codebase. They are now far simpler and perform their job as an RP way of leaving the shift that doesn't involve suicide. They no longer delete traitor items or items in general. They can no longer be used to round remove SSD people without leaving a trail of logs behind.

## Changelog
:cl:
fix: Cryopods can no longer be used to round remove other SSD players.
fix: Cryopods no longer delete traitor objectives under certain edge cases, and can no longer be used to metagame the existence of traitors with steal objectives.
balance: Cryopods now vomit out all the items of players who get vored by them, like beautiful loot pinatas.
/:cl:

### CODE REVIEW

File: tgui/packages/tgui/interfaces/CryopodConsole.js [hunk 1]
```diff
@@ -33,7 +30,7 @@ const CrewList = (props, context) => {
       {!frozen_crew.length ? (
         <NoticeBox>No stored crew!</NoticeBox>
       ) : (
-        <Section height={10} fill scrollable>
+        <Section height={"100%"} scrollable>
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/CryopodConsole.js:33]: Section cannot fill if it's inside the collapsible. You need to implement this in some other way if you want to have a scrollbar.

**Comment by celotajstg** [comment]
why drop items? having an ID drop isn't a good idea

**Comment by wesoda25** [comment]
Because 99% of suicides don't delete items

File: tgui/packages/tgui/interfaces/CryopodConsole.js (hunk 1 above)
Comment (Timberpoes) [reply tgui/packages/tgui/interfaces/CryopodConsole.js:33]: When I didn't have height="100%" the section had a height of 0 for whatever reason and wouldn't render properly.

I'll look into this more when I can, possibly not needing a collapsible anymore as it's only showing that one piece of info now.

Comment (Timberpoes) [reply tgui/packages/tgui/interfaces/CryopodConsole.js:33]: No crew:
image

With crew:
image

**Comment by Timberpoes** [comment]
> 
> 
> why drop items? having an ID drop isn't a good idea
> 
> just delete the items

If you don't want your ID taken when round removing yourself in the HRP tube, hand it in to your Head of Staff, the HoP, the Captain or just throw it into space.

**Comment by Mothblocks** [comment]
because i told them to 🙂

File: code/game/machinery/cryopod.dm [hunk 2]
```diff
@@ -305,29 +249,23 @@ GLOBAL_LIST_EMPTY(cryopod_computers)
...
 
-	for(var/obj/item/item in mob_occupant.GetAllContents())
-		if(item.loc.loc && (item.loc.loc == loc || item.loc.loc == control_computer))
-			continue // means we already moved whatever this thing was in
-			// I'm a professional, okay
-
-		if(!should_preserve_item(item))
+#if MIN_COMPILER_VERSION >= 514
+	#warn Please replace the loop below this warning with an `as anything` loop.
+#endif
```
Comment (Mothblocks) [inline code/game/machinery/cryopod.dm:254]: :pensive:

File: code/game/machinery/cryopod.dm [hunk 3]
```diff
@@ -345,52 +283,50 @@ GLOBAL_LIST_EMPTY(cryopod_computers)
...
 			to_chat(user, "<span class='danger'>You can't put [target] into [src]. [target.p_theyre(capitalized = TRUE)] online.</span>")
 		else
 			to_chat(user, "<span class='danger'>You can't put [target] into [src]. [target.p_theyre(capitalized = TRUE)] conscious.</span>")
 		return
-	else if(target.client)
-		if(alert(target,"Would you like to enter cryosleep?",,"Yes","No") == "No")
-			return
 
-	var/generic_plsnoleave_message = " Please adminhelp before leaving the round, even if there are no administrators online!"
+	if(target == user && (tgalert(target, "Would you like to enter cryosleep?", "Enter Cryopod?", "Yes", "No") == "No"))
```
Comment (Mothblocks) [inline code/game/machinery/cryopod.dm:293]:
```suggestion
	if(target == user && (tgalert(target, "Would you like to enter cryosleep?", "Enter Cryopod?", "Yes", "No") != "Yes"))
```

`tgalert` has the potential to return null in specific cases.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Review by Mothblocks — APPROVED** [review]
DM is good, waiting on tgui pass.

File: code/game/machinery/cryopod.dm [hunk 4]
```diff
@@ -305,29 +249,23 @@ GLOBAL_LIST_EMPTY(cryopod_computers)
...
-			// I'm a professional, okay
-
-		if(!should_preserve_item(item))
+#if MIN_COMPILER_VERSION >= 514
+	#warn Please replace the loop below this warning with an `as anything` loop.
+#endif
+	for(var/mob_content in mob_occupant)
+		var/obj/item/item_content = mob_content
+		if(!istype(item_content))
 			continue
```
Comment (TemporalOroboros) [inline code/game/machinery/cryopod.dm:258]: How does this interact with storage implants and the like?

Comment (Mothblocks) [reply code/game/machinery/cryopod.dm:258]: im curious to see if it spills out all ur organs

**Pull request merged**